lib/carrierwave/sanitized_file.rb

```diff
@@ -330,9 +330,16 @@ def existing_content_type
 
     def mime_magic_content_type
       if path
-        File.open(path) do |file|
-          MimeMagic.by_magic(file).try(:type) || 'invalid/invalid'
+        type = File.open(path) do |file|
+          MimeMagic.by_magic(file).try(:type)
         end
+
+        if type.nil?
+          type = ::MiniMime.lookup_by_filename(path).try(:content_type)
+          type = 'invalid/invalid' unless type.nil? || type.start_with?('text/')
```

Comment:
I'm curious why you didn't go with your [original approach](https://github.com/carrierwaveuploader/carrierwave/issues/2424#issuecomment-618238080) for `image/*` check only? There are a lot of other options like `application/*` or `audio/*` for which the issue will still be relevant

Reply:
The actual strategy on carrierwave is giving high confidence in the result of `MimeMagic.by_magic`:  
* If it returns something : it's valid
* If it returns nothing : it's invalid

The approach I suggested in [#2424](https://github.com/carrierwaveuploader/carrierwave/issues/2424#issuecomment-618238080) doesn't trust the result and still considers missing types as invalid:  
* If it returns an image : we double-check the type from the filename
* If it's present but anything other than an image : it's valid
* If it returns nothing : it's still invalid

The final solution tries to keep confidence in `MimeMagic.by_magic` but considers the nothing case:
* If it returns something : it's still valid.
* If it returns nothing : we double-check the type from the file name:
  * if it's still missing : it's just something unknown so let's return `nil`
  * if it looks like text : the types don't match but there is no risk in considering it valid
  * if it's anything else : it looks like type spoofing so let's return invalid type


I guess images, audio or video files not detected as such by `MimeMagic.by_magic` should be considered at risk as they can then be processed by sub-programs, added and played from HTML pages.

I've mixed feelings about `application/*` because it covers a lot of cases.